### ISSUE TITLE
Document the engine

### ISSUE THREAD

**Issue by birbe** [wgpu-mc/wgpu-mc#70]
Related #69 

Add proper descriptions to structs and methods

**Comment by schitcrafter** [comment]
In my opinion, a more high-level overview would be best because it's very hard to piece together everything from individual structs and methods, no matter how good their descriptions are, and it would be useful for newcomers to make some small, quick changes.

**Comment by birbe** [comment]
A high-level overview isn't enough because there's a lot of moving parts and if you're trying to figure out how to say, do entity rendering, it's important that you can understand every piece of the puzzle at a low level

**Comment by schitcrafter** [comment]
Yes, but I think it would be ideal to start with the high-level docs - then I could, for example, figure out some parts of the project in more detail and write in-depth documentation myself. I suggested the high-level docs because the issue itself only talked about descriptions for structs and methods, and IMO that's the wrong place to start (I already tried it a while ago, went horribly)
